cmd/soroban-rpc/internal/util/monitoredRoutine.go

```diff
@@ -0,0 +1,59 @@
+package util
+
+import (
+	"os"
+	"reflect"
+	"runtime"
+	"runtime/debug"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stellar/go/support/log"
+)
+
+type MonitoredRoutineConfiguration struct {
+	Log                *log.Entry
+	ExitProcessOnPanic bool
+	PanicsCounter      prometheus.Counter
+}
+
+// MonitoredRoutine give us the ability to spin a goroutine, with clear upfront definitions on what should be done in the
+// case of an internal panic.
+func MonitoredRoutine(cfg MonitoredRoutineConfiguration, fn func()) {
+	go func() {
+		defer recoverRoutine(&cfg, fn)
+		fn()
+	}()
+}
+
+func recoverRoutine(cfg *MonitoredRoutineConfiguration, fn func()) {
+	recoverRes := recover()
+	if recoverRes == nil {
+		return
+	}
+	if cfg.Log != nil {
```

Comment:
is it worthwhile to print this detailed content to stderr if `cfg.Log == nil` ? 

For deployments, the log is going to console anyways correct? So, wondering if the monitoring routine needs a ref to a logger, rather it could just print all this new info to stderr, then it will be available uniformly at deployment time, as most deployments scrape stdin/stdout of a process.

Reply:
For now, the log output goes to the console. In the future, we might redirect it to log files.. so, to your point -
I think that outputting that to a log is the right call. But outputting that to the stderr is also a good idea.